pilot/pkg/model/authentication.go

```diff
@@ -88,27 +95,60 @@ func (policy *AuthenticationPolicies) addRequestAuthentication(configs []Config)
 	}
 }
 
+func (policy *AuthenticationPolicies) addPeerAuthentication(configs []Config) {
+	for _, config := range configs {
+		policy.peerAuthentications[config.Namespace] =
+			append(policy.peerAuthentications[config.Namespace], config)
+	}
+}
+
 // GetJwtPoliciesForWorkload returns a list of JWT policies matching to labels.
 func (policy *AuthenticationPolicies) GetJwtPoliciesForWorkload(namespace string,
+	workloadLabels labels.Collection) []*Config {
+	extractor := func(cfg *Config) map[string]string {
+		return cfg.Spec.(*v1beta1.RequestAuthentication).GetSelector().GetMatchLabels()
+	}
+	return getConfigsForWorkload(policy.requestAuthentications, policy.rootNamespace, extractor, namespace, workloadLabels)
+}
+
+// GetPeerAuthenticationsForWorkload returns a list of peer authentication policies matching to labels.
+func (policy *AuthenticationPolicies) GetPeerAuthenticationsForWorkload(namespace string,
+	workloadLabels labels.Collection) []*Config {
+	extractor := func(cfg *Config) map[string]string {
+		return cfg.Spec.(*v1beta1.PeerAuthentication).GetSelector().GetMatchLabels()
+	}
+	return getConfigsForWorkload(policy.peerAuthentications, policy.rootNamespace, extractor, namespace, workloadLabels)
+}
+
+// GetRootNamespace return root namespace that is tracked by the policy object.
+func (policy *AuthenticationPolicies) GetRootNamespace() string {
+	return policy.rootNamespace
+}
+
+type extractSelector func(*Config) map[string]string
```

Comment:
func comment for type defintion, what map[string]string stands for

Reply:
Removed.